include/dcgp/algorithms/es4cgp.hpp

```diff
@@ -0,0 +1,284 @@
+        std::transform(best_x.begin(), best_x.end(), best_xu.begin(),
+                       [](double a) { return boost::numeric_cast<unsigned>(a); });
+        // A contiguous vector of chromosomes/fitness vectors for pagmo::bfe input\output is allocated here.
+        pagmo::vector_double dvs(NP * dim);
+        pagmo::vector_double fs(NP * n_obj);
+        // Main loop
+        for (decltype(m_gen) gen = 1u; gen <= m_gen; ++gen) {
+            // Logs and prints (verbosity modes > 1: a line is added every m_verbosity generations)
+            if (m_verbosity > 0u) {
+                // Every m_verbosity generations print a log line
+                if (gen % m_verbosity == 1u || m_verbosity == 1u) {
+                    // Every 50 lines print the column names
+                    if (count % 50u == 1u) {
+                        pagmo::print("\n", std::setw(7), "Gen:", std::setw(15), "Fevals:", std::setw(15), "Best:\n");
+                    }
+                    pagmo::print(std::setw(7), gen - 1, std::setw(15), prob.get_fevals() - fevals0, std::setw(15),
+                                 best_f, '\n');
+                    ++count;
+                    // Logs
+                    m_log.emplace_back(gen - 1, prob.get_fevals() - fevals0, best_f);
+                }
+            }
+
+            // 1 - We generate new NP individuals mutating the best and we write on the dvs for pagmo::bfe to evaluate
+            // their fitnesses.
+            for (decltype(NP) i = 0u; i < NP; ++i) {
+                // To mutate active chromosomes we use a copy of the cgp of the UDP.
+                cgp.set(best_xu);
+                cgp.mutate_active(m_mut_n);
+                std::vector<unsigned> mutated_x = cgp.get();
+                std::transform(mutated_x.begin(), mutated_x.end(), dvs.data() + i * dim,
+                               [](unsigned a) { return boost::numeric_cast<double>(a); });
+            }
+            // 3 - We compute their fitnesses calling the bfe
+            fs = m_bfe(prob, dvs);
+            // Check if ftol exit condition is met
+            if (best_f < m_ftol) {
+                if (m_verbosity > 0u) {
+                    pagmo::print(std::setw(7), gen, std::setw(15), prob.get_fevals() - fevals0, std::setw(15), best_f,
+                                 '\n');
+                    ++count;
+                    // Logs
+                    m_log.emplace_back(gen, prob.get_fevals() - fevals0, best_f);
+
+                    pagmo::print("Exit condition -- ftol < ", m_ftol, "\n");
+                }
+                update_pop(pop, dvs, fs, best_f, best_xu, NP, dim);
+                return pop;
+            }
+            // 4 - We reinsert the mutated individuals in the population if their fitness is
+            // less than, or equal, to the one from the parent.
+            for (decltype(NP) i = 0u; i < NP; ++i) {
+                if (pagmo::detail::less_than_f(fs[i], best_f)) {
```

Comment:
I can copy it into a dcgp detail namespace for the time being ... or leave it and wait for the next point release of pagmo?

Reply:
Nah no need to do anything here, we cannot do it in a point release anyway as it changes the binary interface. Let's leave this for later, we'll modify DCGP after the pagmo changes (it will stop compiling anyway :) )